### PR TITLE
feat: mobile replay in usage

### PR DESCRIPTION
**Pull request by pauldambra**
Let's 
 
* report mobile and web replay usage separately
* not count mobile towards billable while we're still iterating so heavily

I _think_ this does that

### CODE REVIEW

File: posthog/tasks/usage_report.py [hunk 1]
```diff
@@ -500,15 +504,23 @@ def get_teams_with_event_count_by_name(begin: datetime, end: datetime) -> List[T
...
         SELECT team_id, count(distinct session_id) as count
-        FROM session_replay_events
-        WHERE min_first_timestamp BETWEEN %(begin)s AND %(end)s
-        AND session_id NOT IN (
+        FROM (
+            SELECT any(team_id) as team_id, session_id
+            FROM session_replay_events
+            WHERE min_first_timestamp BETWEEN %(begin)s AND %(end)s
+            GROUP BY session_id
+            -- exclude mobile recordings here
```
Comment (marandaneto) [inline posthog/tasks/usage_report.py:520]: why do we exclude mobile recordings here? shouldn't the `where` with `snapshot_source` include only actually what you want? either mobile or web.

Comment (pauldambra) [reply posthog/tasks/usage_report.py:520]: ah, rushing to get the PR In before doing the school run.

Yep, the comment is out of sync with reality...

Comment (pauldambra) [reply posthog/tasks/usage_report.py:520]:
```suggestion
```

File: posthog/tasks/usage_report.py [hunk 2]
```diff
@@ -531,13 +543,20 @@ def get_teams_with_recording_count_in_period(begin: datetime, end: datetime) ->
...
+def get_teams_with_recording_count_total(snapshot_source: Literal["mobile", "web"]) -> List[Tuple[int, int]]:
     result = sync_execute(
         """
         SELECT team_id, count(distinct session_id) as count
-        FROM session_replay_events
+        FROM (
+            SELECT any(team_id) as team_id, session_id
+            FROM session_replay_events
+            GROUP BY session_id
+            -- exclude mobile recordings here
```
Comment (pauldambra) [inline posthog/tasks/usage_report.py:554]:
```suggestion
```

File: posthog/tasks/test/test_usage_report.py [hunk 3]
```diff
@@ -56,6 +56,81 @@
 logger = structlog.get_logger(__name__)
 
 
+def _setup_replay_data(team_id: int, include_mobile_replay: bool) -> None:
+    # recordings in period  - 5 sessions
```
Comment (pauldambra) [inline posthog/tasks/test/test_usage_report.py:60]: we were looping here to add "multiple snapshots" but we don't store snapshots in replay table any more so we can avoid the inner loop and make this a _little_ cheaper to run at least 😊

File: posthog/tasks/test/test_usage_report.py [hunk 4]
```diff
@@ -658,6 +692,38 @@ def test_unlicensed_usage_report(self, mock_post: MagicMock, mock_client: MagicM
...
+
+        period = get_previous_day()
+        period_start, period_end = period
+
+        all_reports = _get_all_usage_data_as_team_rows(period_start, period_end)
+        report = _get_team_report(all_reports, self.team)
+
+        assert all_reports["teams_with_recording_count_total"] == {self.team.pk: 16}
+        assert report.recording_count_in_period == 5
+        assert report.recording_count_total == 16
```
Comment (pauldambra) [inline posthog/tasks/test/test_usage_report.py:708]:
```suggestion
        assert report.recording_count_total == 16
    
        assert report.mobile_recording_count_in_period == 1
        assert report.mobile_recording_count_total == 2
```

**Review by marandaneto — APPROVED** [review]

File: posthog/tasks/usage_report.py [hunk 5]
```diff
@@ -689,6 +706,7 @@ def has_non_zero_usage(report: FullUsageReport) -> bool:
         report.event_count_in_period > 0
         or report.enhanced_persons_event_count_in_period > 0
         or report.recording_count_in_period > 0
+        # explicitly not including mobile_recording_count_in_period for now
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/tasks/usage_report.py:709]: Yep, I'm guessing we only want to count not-free things

It's called out in the comment here in the hope @PostHog/team-growth correct me :D

[the remark this answers is not in the document]
Comment (marandaneto) [reply posthog/tasks/usage_report.py:709]: oh I see, that makes sense, well it's not "free" yet but the goal is to be for a while.

[the remark this answers is not in the document]
Comment (raquelmsmith) [reply posthog/tasks/usage_report.py:709]: Yeah this just defines whether or not we send the report to billing. You'll still get the org usage report that you can query against in PostHog events.

**Review by raquelmsmith — APPROVED** [review]
Looks good to me!

**Pull request merged**